packages/block-editor/src/components/iframe/index.js

```diff
@@ -195,19 +201,23 @@ function Iframe( {
 		};
 	}, [] );
 
-	const headRef = useRefEffect( ( element ) => {
+	useEffect( () => {
+		if ( ! ( head && scripts ) ) {
+			return;
+		}
 		scripts
 			.reduce(
 				( promise, script ) =>
-					promise.then( () => loadScript( element, script ) ),
+					promise.then( () => loadScript( head, script ) ),
 				Promise.resolve()
 			)
 			.finally( () => {
 				// When script are loaded, re-render blocks to allow them
 				// to initialise.
 				forceRender();
 			} );
-	}, [] );
+	}, [ head, scripts ] );
```

Comment:
This is possibly a bit risky, as if `scripts` changes, it'd mean duplicate scripts being attached. It might be safer to remove the `scripts` dep, or keep track of already loaded scripts.

Reply:
Since we have the `id` attribute of the script, could we remove the existing script before creating a new `<script>` element? Or just skip creating that script, whichever makes more sense.

Reply:
I changed it back to work the way it did in `trunk`.

Your suggestion sounds good, but I think it's best to address the script loading in a separate PR, as it's unrelated to the bug.